src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -240,15 +240,23 @@ protected override NuGetIndexSearcher CreateSearcher(IndexReader reader)
 
                 // There are four flavors of Latest/Listed filter to reflect all the possible combinations.
 
-                var h00 = new LatestListedHandler(includeUnlisted: false, includePrerelease: false);
-                var h01 = new LatestListedHandler(includeUnlisted: false, includePrerelease: true);
-                var h10 = new LatestListedHandler(includeUnlisted: true, includePrerelease: false);
-                var h11 = new LatestListedHandler(includeUnlisted: true, includePrerelease: true);
-
-                indexReaderProcessor.AddHandler(h00);
-                indexReaderProcessor.AddHandler(h01);
-                indexReaderProcessor.AddHandler(h10);
-                indexReaderProcessor.AddHandler(h11);
+                var h000 = new LatestListedHandler(includeUnlisted: false, includePrerelease: false, includeSemver2: false);
+                var h010 = new LatestListedHandler(includeUnlisted: false, includePrerelease: true, includeSemver2: false);
+                var h100 = new LatestListedHandler(includeUnlisted: true, includePrerelease: false, includeSemver2: false);
+                var h110 = new LatestListedHandler(includeUnlisted: true, includePrerelease: true, includeSemver2: false);
+                var h001 = new LatestListedHandler(includeUnlisted: false, includePrerelease: false, includeSemver2: true);
+                var h011 = new LatestListedHandler(includeUnlisted: false, includePrerelease: true, includeSemver2: true);
+                var h101 = new LatestListedHandler(includeUnlisted: true, includePrerelease: false, includeSemver2: true);
+                var h111 = new LatestListedHandler(includeUnlisted: true, includePrerelease: true, includeSemver2: true);
+
+                indexReaderProcessor.AddHandler(h000);
```

Comment:
This code must be backwards compatible, otherwise we will be in a world of pain during deployment/rollback. Meaning, in case the index doesn't contain the new SemVer 2 properties, the code should fallback to default and just work. 
Is it?